### PR TITLE
done, add linenum on image

### LINKED ISSUE #364: Show line number in source code image
Can be drawing ASCII-art line numbering like this:

```
 1  | #include <bits/stdc++h>
 2  | using namespace std;
 3  |
 4  | int main() {
 5  |     int A, B;
 6  |     cin >> A >> B;
 7  |
 8  |     int C = A + B;
 9  |     cout << C << endl;
10  | }
```

### PR DESCRIPTION
**Pull request by FwP-IDN**
Fixes #364

### CODE REVIEW

File: judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java [hunk 1]
```diff
@@ -72,6 +72,8 @@ public static Response buildImageResponseFromText(String text, Date lastModified
         int charHeight = 16;
 
         String[] textList = text.split("\\r?\\n");
+        int maxDigitLineNumber = String.valueOf(textList.length).length();
+        String lineTemplate = String.format(" %%%dd  | %%s", maxDigitLineNumber);
```
Comment (fushar) [inline judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java]:
```suggestion
        String lineTemplate = String.format(" %%%dd | %%s", maxDigitLineNumber);
```

(it looks better) 😄

File: judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java [hunk 2]
```diff
@@ -85,8 +87,9 @@ public static Response buildImageResponseFromText(String text, Date lastModified
         g2d.fillRect(0, 0, img.getWidth(), img.getHeight());
         g2d.setColor(Color.BLACK);
         int nextLinePosition = margin;
-        for (String s : textList) {
-            g2d.drawString(s.replaceAll("\t", "    "), 0, nextLinePosition);
+        for (int i = 0; i < textList.length; i++) {
+            String s = textList[i];
+            g2d.drawString(String.format(lineTemplate, i+1, s.replaceAll("\t", "    ")), 0, nextLinePosition);
```
Comment (fushar) [inline judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java]:
```suggestion
            g2d.drawString(String.format(lineTemplate, i + 1, s.replaceAll("\t", "    ")), 0, nextLinePosition);
```

(not sure why it was not caught by checkstyle)

**Comment by fushar** [comment]
While we're at it, can we make the line numbers and `|` separators to have different color to make it nicer? (e.g. blue) 😄

**Review by fushar — CHANGES_REQUESTED** [review]

File: judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java (hunk 2 above)
Comment (FwP-IDN) [reply judgels-backends/judgels-commons/judgels-service-core/src/main/java/judgels/service/ServiceUtils.java:92]: I forgot to install Checkstyle IDEA, hahahahha

**Comment by FwP-IDN** [comment]
> While we're at it, can we make the line numbers and `|` separators to have different color to make it nicer? (e.g. blue) 😄

done

wis

**Review by fushar — APPROVED** [review]

**Pull request merged**